### DISCUSSION TITLE
NUT errore connessione

### DISCUSSION THREAD

**Discussion by pisanialessandro** [indomus/forum#2432] [HUB: Home Assistant]
Ciao, ho acquistato UPS Tecnoware 1100 e seguito la guida per configurare NUT su Mini PC con Docker su Linux Debian.
Tutto funziona a meraviglia come da guida.

Quando vado ad installare su Home Assistant la nuova integrazione NUT, inserisco i dati di connessione:

**localhost**, 3493, user e psw  oppure
**a0d7b954-nut**, 3493, user e psw  oppure
**IP Mini pc**, 3493, user e psw  

Home assistant mi dice che è impossibile connettersi.

come posso risolvere?

**Comment by KC-inDomus** [comment]
ti sei messo in ascolto con `docker logs nut-upsd -f` per vedere cosa riporta il container nut quando provi a collegarti da HA?

**Reply by pisanialessandro** [reply]
niente, il log non reagisce quando provo a collegarmi.

Questo è il log da mini pc di corretto funzionamento:
```
   0.000000	UPS: tecnoware@localhost (master) (power value 1)
   0.000031	debug level is '1'
   0.000433	Trying to connect to UPS [tecnoware@localhost]
   0.000909	Logged into UPS tecnoware@localhost
```
 
Mi viene il dubbio che non si riesca a raggiungere il server NUT perché su Docker.
Ho provato questo prompt, non so se è giusto e non sono molto pratico di docker.

```
deb@debian:~$ telnet localhost 3493
Trying ::1...
Trying 127.0.0.1...
telnet: Unable to connect to remote host: Connection refused
```

Grazie

**Reply by KC-inDomus** [reply]
Qual è la configurazione del container docker che hai messo su?

**Reply by snakuzzo** [reply]
come già chiesto un paio di giorni fa, condividi i dettagli su come hai creato il container.
Se non risponde sulla sua porta, probabilmente non hai configurato correttamente il container

**Reply by pisanialessandro** [reply]
Questo è il comando di creazione:

`docker run --name nut-upsd --hostname=nut-upsd --device "/dev/bus/usb/001/002" --env UPS_NAME="tecnoware" --env UPS_DRIVER="blazer_usb" --env API_USER="master" --env API_PASSWORD="XXXX" --env ADMIN_PASSWORD="YYY" --detach --privileged --restart always upshift/nut-upsd`

**Reply by pisanialessandro** [reply]
quando provo ad installare su HA NUT uso come utente master (come indicato nel comando di creazione) e relativa password.

**Reply by snakuzzo** [reply]
Non funzionerà mai così...devi fare anche il bind della porta o configurare network host

**Reply by pisanialessandro** [reply]
Scusate l'ignoranza ma non ho idea come farlo, ho seguito l'ottima guida passo passo:

https://indomus.it/guide/gestire-un-ups-con-raspberry-pi-tramite-network-ups-tools-nut-con-docker-su-raspbian/

vi ringrazio in anticipo!

**Reply by pisanialessandro** [reply]
Questa è la lista dei container
![image](https://user-images.githubusercontent.com/17181327/232538317-b729c5a2-47a8-4131-bec7-f52fefe74f8b.png)

**Reply by snakuzzo** [reply]
Sì ma io non ti ho chiesto i container. Ti ho scritto qual è il problema che hai. Se ci fai caso hai dimenticato

```
-p 3493:3493
```

**Reply by pisanialessandro** [reply]
Cavoli, hai ragione!!

Adesso funziona a meraviglia, grazie.
